pkg/3scale/amp/operator/redis_options_provider.go

```diff
@@ -272,3 +280,44 @@ func (r *RedisOptionsProvider) setTopologySpreadConstraints() {
 		r.options.BackendRedisTopologySpreadConstraints = r.apimanager.Spec.Backend.RedisTopologySpreadConstraints
 	}
 }
+
+func (r *RedisOptionsProvider) systemRedisPodTemplateAnnotations() (map[string]string, error) {
+	annotations := make(map[string]string)
+	for k, v := range r.apimanager.Spec.System.RedisAnnotations {
+		annotations[k] = v
+	}
+	resourceVersion, err := r.redisConfigMapResourceVersion()
+	if err != nil {
+		return nil, err
+	}
+	annotations["generationID"] = resourceVersion
+	return annotations, nil
+}
+
+func (r *RedisOptionsProvider) backendRedisPodTemplateAnnotations() (map[string]string, error) {
+	annotations := make(map[string]string)
+	for k, v := range r.apimanager.Spec.Backend.RedisAnnotations {
+		annotations[k] = v
+	}
+	resourceVersion, err := r.redisConfigMapResourceVersion()
+	if err != nil {
+		return nil, err
+	}
+	annotations["generationID"] = resourceVersion
+	return annotations, nil
+}
+
+func (r *RedisOptionsProvider) redisConfigMapResourceVersion() (string, error) {
+	// get resourceVersion from CM and use it as Annotation "generationID" for redis Pods
+	key := client.ObjectKey{
+		Namespace: r.namespace,
+		Name:      "redis-config",
+	}
+	cm := &v1.ConfigMap{}
+	err := r.client.Get(context.Background(), key, cm)
+	if err != nil {
+		fmt.Printf("Failed to get ConfigMap: %v\n", err)
```

Comment:
use logger or let caller to log the error.

Reply:
Done. Thank you